build.rs

```diff
@@ -3,61 +3,170 @@
+    );
 
-    if cfg!(feature = "thirsty") {
-        enabled_responses += &serialise_responses("thirsty", &responses);
+    let mut vars = String::new();
+    for (name, var) in &config.vars {
+        if var.spiciness > Spiciness::CONFIGURED {
+            continue;
+        }
+
+        let env_key = var
+            .env_key
+            .clone()
+            .unwrap_or_else(|| format!("CARGO_MOMMYS_{}S", name.to_uppercase()));
+        let defaults = &var.defaults;
+        let _ = write!(
+            vars,
+            "Var {{ env_key: {env_key:?}, defaults: &{defaults:?} }},"
+        );
     }
-    if cfg!(feature = "yikes") {
-        enabled_responses += &serialise_responses("yikes", &responses);
+
+    let pattern = Regex::new(r"\{\w+\}").unwrap();
+    let mut responses = String::new();
+    for (name, mood) in &config.moods {
+        if mood.spiciness > Spiciness::CONFIGURED {
+            continue;
+        }
+
+        let parse_response = |text: &str, out: &mut String| {
+            let _ = write!(out, "&[");
+
+            let mut prev = 0;
+            for var in pattern.find_iter(text) {
+                let var_name = &var.as_str()[1..var.len() - 1];
+                let var_idx = match var_map.get(var_name) {
+                    Some(&i) => {
+                        assert!(
+                            config.vars[i].1.spiciness <= Spiciness::CONFIGURED,
+                            "{{{var_name}}} is too spicy!"
+                        );
+                        i
+                    }
+                    None => panic!("undeclared variable {{{var_name}}}"),
+                };
+
+                let Range { start, end } = var.range();
+                let prev_chunk = &text[prev..start];
+                prev = end;
+
+                let _ = write!(out, "Chunk::Text({prev_chunk:?}), Chunk::Var({var_idx}), ");
+            }
+
+            let _ = write!(out, "Chunk::Text({:?})],", &text[prev..],);
+        };
+
+        let _ = write!(responses, "Mood {{ name: {name:?}, positive: &[");
+        for response in &mood.positive {
+            parse_response(&response, &mut responses)
+        }
+        let _ = write!(responses, "], negative: &[");
+        for response in &mood.negative {
+            parse_response(&response, &mut responses)
+        }
+        let _ = write!(responses, "] }},");
     }
 
     fs::write(
         dest_path,
         format!(
-            r#"const RESPONSES: &[(&str, &[&[&str]])] = &[
-                {enabled_responses}
-            ];"#
+            r"
+            const CONFIG: Responses<'static> = Responses {{
+                vars: &[{vars}],
+                moods: &[{responses}],
+            }};
+            const MOOD: &'static Var<'static> = &CONFIG.vars[0];
+            const EMOTE: &'static Var<'static> = &CONFIG.vars[1];
+            const PRONOUN: &'static Var<'static> = &CONFIG.vars[2];
+            const ROLE: &'static Var<'static> = &CONFIG.vars[3];
```

Comment:
For the preceding reasons, to guarantee these all have one location in memory, so we can reason about their placement, mommy should use `static` instead. Mommy's a bit fluffy and gets a bit of friction, I guess, huh~? Watch out for zaps:zap: 
```suggestion
            static CONFIG: Responses<'static> = Responses {{
                vars: &[{vars}],
                moods: &[{responses}],
            }};
            static MOOD: &'static Var<'static> = &CONFIG.vars[0];
            static EMOTE: &'static Var<'static> = &CONFIG.vars[1];
            static PRONOUN: &'static Var<'static> = &CONFIG.vars[2];
            static ROLE: &'static Var<'static> = &CONFIG.vars[3];
```

Reply:
> Watch out for zaps⚡
my brain: what the fuck does this have to do with pointer zapping 😱

I think I'm going to just. not use pointer equality. This doesn't actually matter (as far as perf goes) at the LLVM level, at the end of the day...

Wait, why am i thinking about the perf of a novelty D/s build system plugin?